### ISSUE TITLE
There should be an option to override the attribute type like PyCaret

### ISSUE THREAD

**Issue by akhil14shukla** [8080labs/ppscore#52]
Numerical attributes not always point to regression and could be categorical attributes, where a Classification model should be used.
PyCaret offers such features, where we can list numerical and categorical features. Something of that kind will greatly increase the usability of this library.

**Comment by FlorianWetschoreck** [comment]
Currently, this can be achieved via adjusting the data type(s) of the respective column(s).

What is your preferred/suggested way of providing this information? And how does this compare to change the data type?

My first thought is to pass something like
- `categoric_target=True` for pps.score
- and `categoric_targets=["column_a", ...]` for pps.predictors/matrix

Maybe passing `categoric_columns=["column_a", ...]` is even better. This could be used for score, predictors and matrix.

However, I am still hesitant because the same thing can be achieved via transforming the data type in the dataframe. Beyond being a different syntax, I do not see a difference. Maybe it improves discoverability and clarity about this issue?

**Comment by akhil14shukla** [comment]
I think creating a basic function that passes `categoric_columns = ["column_a", ...]`, is better, as we will have better control, and won't be needing to change the data type (I hope you mean converting to object or string). Changing the data type will create problems while training some models or plotting data, and we will again need to revert the data type to numerical.
Also, passing the categorical columns is reasonable for small datasets with a few columns/features, but it will be very troublesome for large ones. So, I don't think incorporating this feature of mentioning attribute type is futile, it is very helpful. But we also need a better judge of an attribute type for large datasets. 
I faced this issue, when my column had only two values 0 and 1, and it was performing regression. 
Off the top of my head, a basic solution for the above problems could be:  
* If the column data is of type _int_ and has less than 10 unique values, then the algorithm should assume it as a categorical attribute
* In addition to the above, an option to explicitly pass the columns of categorical or numerical type

**Comment by FlorianWetschoreck** [comment]
I agree that there a `autodetect_categorical_columns=True` flag might be helpful if there are many columns.
In the beginning, we had something per default but this did lead to more confusion than it did help. Thus, such a flag should be `False` by default.

**Comment by akhil14shukla** [comment]
Yes, it should surely depend on the user needs. I created a basic version of the PPS that I was talking about, in which we can pass categorical columns, it is working fine so far [Link to file](https://github.com/akhil14shukla/PyCustom/blob/master/ppscore.py)
